tests/unittests/tests-uri_parser/tests-uri_parser.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (C) 2020 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ */
+
+#include <stdio.h>
+#include "embUnit.h"
+
+#include "uri_parser.h"
+
+#include "unittests-constants.h"
+#include "tests-uri_parser.h"
+
+#define VEC(u, s, us, h, po, pa, q, e)                                  \
+    { .uri = u, .scheme = s, .userinfo = us, .host = h, .port = po,     \
+      .path = pa, .query = q, .expected = e}
+
+#define VEC_CHECK(comp)                                                 \
+    do {                                                                \
+        if (ures.comp == NULL) {                                        \
+            TEST_ASSERT(validate_uris[i].comp[0] == '\0');              \
+        }                                                               \
+        else {                                                          \
+            TEST_ASSERT_EQUAL_INT(strlen(validate_uris[i].comp),        \
+                                  ures.comp##_len);                     \
+            TEST_ASSERT_EQUAL_INT(0,                                    \
+                                  memcmp(ures.comp,                     \
+                                         validate_uris[i].comp,         \
+                                         strlen(validate_uris[i].comp))); \
```

Comment:
Ahhhh this is what was confusing me. Can we use maybe

```suggestion
            TEST_ASSERT_EQUAL_STRING(ures.comp,                         \
                                     validate_uris[i].comp);            \
```

here?

Reply:
I tried it myself. Since the strings in `ures` are not zero-terminated: no.

Reply:
Unfortunately this doesn't work, because `ures.comp` is not `Zero`-terminated. Since `TEST_ASSERT_EQUAL_STRING` is using `strcmp()` underneath, it just walks over the valid memory region .. since there is no strncmp test macro, I went with memcmp.

Reply:
Why `memcmp` and not `strncmp`?